src/wp-includes/class-wp-locale-switcher.php

```diff
@@ -196,11 +196,12 @@ private function load_translations( $locale ) {
 		load_default_textdomain( $locale );
 
 		foreach ( $domains as $domain ) {
+			// The default text domain is handled by `load_default_textdomain()`.
 			if ( 'default' === $domain ) {
 				continue;
 			}
 
-			unload_textdomain( $domain );
+			unload_textdomain( $domain, true );
```

Comment:
```suggestion
			// Unload current text domain but allow them to be reloaded
			// after switching back or to another locale.
			unload_textdomain( $domain, true );
```

Some docs so you don't have to check what `true` does.